query/filter/key_builder.go

```diff
@@ -124,7 +119,7 @@ func NewKeyBuilder[F fieldable](composer KeyComposer[F], primaryKey bool) *KeyBu
 // On each level multiple keys can be formed because the user can specify ranges. The builder is not deciding the logic
 // of key generation, the builder is simply traversing on the filters and calling compose where the logic resides.
 // If the build is for a primary key, the query plans are merged into a single plan.
-func (k *KeyBuilder[F]) Build(filters []Filter, userDefinedKeys []F) ([]QueryPlan, error) {
+func (k *KeyBuilder) Build(filters []Filter, userDefinedKeys []*schema.QueryableField) ([]QueryPlan, error) {
```

Comment:
Lets rename `userDefinedKey` to indexedKeys this will make more sense here now.

Reply:
`Build` calls `Compose`, should we change all parameters named `userDefinedKeys` to `indexedKeys`? Does that make sense?

Reply:
Yeah. I'm still not sure that is the best name. Hold off on that until last and we can decide.